plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-slider/edit.tsx

```diff
@@ -3,25 +3,70 @@
  */
 import clsx from 'clsx';
 import { __ } from '@wordpress/i18n';
-import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
 import { PanelBody, ToggleControl, Disabled } from '@wordpress/components';
 import { formatPrice, getCurrency } from '@woocommerce/price-format';
+import {
+	useBlockProps,
+	InspectorControls,
+	withColors,
+
+	// @ts-expect-error - no types.
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalColorGradientSettingsDropdown as ColorGradientSettingsDropdown,
+	// @ts-expect-error - no types.
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalUseMultipleOriginColorsAndGradients as useMultipleOriginColorsAndGradients,
+} from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
-import { EditProps } from './types';
+import { colorNames } from './constants';
+import { getHasColorClasses, getStyleColorVars } from '../../utils/colors';
+import type { EditProps } from './types';
+
+const PriceSliderEdit = ( {
+	clientId,
+	context,
+
+	attributes,
+	setAttributes,
+
+	// Custom colors
+	sliderHandle,
+	setSliderHandle,
+	sliderHandleBorder,
+	setSliderHandleBorder,
+	slider,
+	setSlider,
+}: EditProps ): JSX.Element | null => {
+	const {
+		showInputFields,
+		inlineInput,
+
+		customSliderHandle,
+		customSliderHandleBorder,
+		customSlider,
+	} = attributes;
+
+	const { isLoading, price } = context.filterData;
 
-const Edit = ( { attributes, setAttributes, context }: EditProps ) => {
-	const { showInputFields, inlineInput } = attributes;
 	const blockProps = useBlockProps( {
-		className: 'wc-block-product-filter-price-slider',
+		className: clsx( 'wc-block-product-filter-price-slider', {
+			'is-loading': isLoading,
+			...getHasColorClasses( attributes, colorNames ),
+		} ),
+		style: getStyleColorVars(
+			'wc-product-filter-price',
+			attributes,
+			colorNames
+		),
 	} );
 
-	const { isLoading, price } = context.filterData;
+	const colorGradientSettings = useMultipleOriginColorsAndGradients();
 
 	if ( isLoading ) {
-		return 'Loading...';
+		return <>{ __( 'Loading…', 'woocommerce' ) }</>;
```

Comment:
Nice catch!

Reply:
:-D anyway I think this is something we'd like to replace with a better _loading_ implementation

Reply:
> anyway I think this is something we'd like to replace with a better loading implementation

We should. I used the text just as a POC while building the slider but forgot to replace it with a proper loading animation.